L1Trigger/L1THGCal/src/backend/HGCalShowerShape.cc

```diff
@@ -6,7 +6,7 @@
 #include <unordered_map>
 
 HGCalShowerShape::HGCalShowerShape(const edm::ParameterSet& conf)
-    : threshold_(conf.existsAs<double>("shape_threshold") ? conf.getParameter<double>("shape_threshold") : 0.) {}
+  : threshold_(conf.getParameterSet("histoMax_C3d_clustering_parameters").existsAs<double>("shape_threshold") ? conf.getParameterSet("histoMax_C3d_clustering_parameters").getParameter<double>("shape_threshold") : 0.) {}
```

Comment:
The clustering PSet should be passed to the constructor. Otherwise it won't work for other types of clustering algos.